CRM/Core/Config.php

```diff
@@ -589,7 +589,10 @@ public function registerTemplateDirectory(array $templateDirectory): void {
       return;
     }
     if (($templateDirectory['type'] ?? 'Smarty') === 'Smarty') {
-      CRM_Core_Smarty::singleton()->addTemplateDir($templateDirectory['path']);
+      \Civi::$statics['CRM_Core_Smarty']['PATHS'][] = $templateDirectory['path'];
+      if (CRM_Core_Smarty::$_singleton !== NULL) {
```

Comment:
@eileenmcnaughton To your concern about when `CRM/Core/Smarty.php` is loaded, perhaps the guard here should be more like

```php
if (class_exists('CRM_Core_Smarty', FALSE) && CRM_Core_Smarty::isStarted()) {
```

The `class_exists(..., FALSE)` means that it will not autoload the class -- it should only tell if you the class has already been loaded.

Reply:
@totten hmm - I don't understand how to ensure it's loaded first though - since it was all loading up so quickly